### PR TITLE
Reverted PR: Command line --timeout option for server stop

### PR DESCRIPTION
**Pull request by jimblye**
Fixes issue #19924

**New function**
Adds --timeout= option to the ./server stop command.

**Examples:**
```
./server stop --timeout=5     // 5 seconds
./server stop --timeout=5s    // 5 seconds
./server stop --timeout=5m    // 5 minutes
./server stop --timeout=300   // 300 seconds => 5 minutes
```

**Undocumented but valid options:**
```
./server stop --timeout=1000ms  // 1 second
./server stop --timeout=1h      // 1 hour
```

**Invalid**
`./server start --timeout=30          // timeout not supported for anything action other than stop`

**Help**

./server help stop

Usage: ./server stop serverName [options] [JVM options]

	Stop the running instance of the named server.

Options:

    --force
	Skips the quiesce stage before the server is shut down. Normal server
	stop includes a quiesce stage before the server is shut down. The    
	quiesce stage, a period of 30 seconds, allows services to perform    
	pre-shutdown work; for example, inbound listeners are stopped but    
	existing requests are allowed to complete. The --force option has no 
	effect if server stop was already invoked. If you use the --force    
	option, you might see unexpected exceptions in the messages.log file 
	that occur after the server stop command was received by the server.
    --timeout
	Amount of time to allow for the server to stop before giving up.     
	Specify a positive integer followed by a unit of time, which can be  
	minutes (m), or seconds (s). For example, specify 5 minutes as 5m or 
	300s.

**Note:**
The changes to **MetaTypeUtils.java** & **KernelUtils.java** simply move existing methods to avoid a circular dependency.  These methods provide the parsing of the duration syntax 00**h**00**m**00**s**00**ms**

.

### CODE REVIEW

**Comment by jimblye** [comment]
Fixes issue  #19924

**Comment by jimblye** [comment]
#build 
#libby

File: dev/com.ibm.ws.kernel.boot/resources/com/ibm/ws/kernel/boot/resources/LauncherMessages.nlsprops [hunk 1]
```diff
@@ -128,6 +128,10 @@ error.kernelDefFile=CWWKE0023E: The specified kernel definition ({0}) does not e
 error.kernelDefFile.explanation=The specified kernel definition could not be found.
 error.kernelDefFile.useraction=Ensure that the kernel definition resource exists and can be read by the current user.
 
+error.badOptionValue=CWWKE0024E: The value ({0}) for the option ({1}) is not valid.
```
Comment (dmuelle) [inline dev/com.ibm.ws.kernel.boot/resources/com/ibm/ws/kernel/boot/resources/LauncherMessages.nlsprops]:
```suggestion
error.badOptionValue=CWWKE0024E: The ({0}) value for the ({1}) option  is not valid.
```

Comment (dmuelle) [reply dev/com.ibm.ws.kernel.boot/resources/com/ibm/ws/kernel/boot/resources/LauncherMessages.nlsprops]: do the variables need ({double brackets}) ? Unless we expect readability to be an issue, I'd remove the outter (parentheses). If the potential values are especially long or difficult to read, ok as is.

File: dev/com.ibm.ws.kernel.boot/resources/com/ibm/ws/kernel/boot/resources/LauncherMessages.nlsprops [hunk 2]
```diff
@@ -128,6 +128,10 @@ error.kernelDefFile=CWWKE0023E: The specified kernel definition ({0}) does not e
...
+error.badOptionValue.explanation=The specified value is not appropriate for the specified option.
+error.badOptionValue.useraction=Try using the command (server help) for a description of the options.  If the value is a number, ensure it is in the valid range.
```
Comment (dmuelle) [inline dev/com.ibm.ws.kernel.boot/resources/com/ibm/ws/kernel/boot/resources/LauncherMessages.nlsprops]:
```suggestion
error.badOptionValue.useraction=Run the `server help` command for a description of the options.  If the value is a number, ensure it is in the valid range.
```

File: dev/com.ibm.ws.kernel.boot/resources/com/ibm/ws/kernel/boot/resources/LauncherMessages.nlsprops [hunk 3]
```diff
@@ -141,6 +145,11 @@ warning.singleServer=CWWKE0027W: Only one server may be specified on the command
   subsequent names will be ignored (server={0}, ignored={1}).
 warning.singleServer.explanation=You may specify only one server on the command line. The first server name has been processed; the remaining names have been ignored.
 warning.singleServer.useraction=No action is required. Specify only one server to avoid this warning.
+
+error.optionNotApplicableToAction=CWWKE0028E: The option ({0}) is not applicable for action ({1}).
```
Comment (dmuelle) [inline dev/com.ibm.ws.kernel.boot/resources/com/ibm/ws/kernel/boot/resources/LauncherMessages.nlsprops]:
```suggestion
error.optionNotApplicableToAction=CWWKE0028E: The ({0}) option is not applicable for the ({1}) action.
```

File: dev/com.ibm.ws.kernel.boot/resources/com/ibm/ws/kernel/boot/resources/LauncherMessages.nlsprops [hunk 4]
```diff
@@ -141,6 +145,11 @@ warning.singleServer=CWWKE0027W: Only one server may be specified on the command
...
+error.optionNotApplicableToAction.explanation=The specified option is not allowed for the specified action.
+# Translate the word action in the command.  Enclose in < >.  User is to substitute the appropriate action, like "start", "stop", etc...
+error.optionNotApplicableToAction.useraction=Try using the command (server <action> help) for valid options.
```
Comment (dmuelle) [inline dev/com.ibm.ws.kernel.boot/resources/com/ibm/ws/kernel/boot/resources/LauncherMessages.nlsprops]:
```suggestion
error.optionNotApplicableToAction.useraction=Run the `server <action> help` command  to get information about valid options for the action.
```

File: dev/com.ibm.ws.kernel.boot/resources/com/ibm/ws/kernel/boot/resources/LauncherMessages.nlsprops [hunk 5]
```diff
@@ -710,7 +719,7 @@ warning.earlyRelease=CWWKE0953W: This version of {0} is an unsupported early rel
 warning.earlyRelease.explanation=New features that are not fully tested are made available in early releases to elicit feedback before they are included in a supported release.
 warning.earlyRelease.useraction=Do not use an early release version on production systems, since early release versions are not supported.
 
-error.invalidPhaseName=CWWKE0954E: The "{0}" checkpoint phase specified is empty or unknown.
+error.invalidPhaseName=CWWKE0954E: The ({0}) checkpoint phase specified is empty or unknown.
```
Comment (dmuelle) [inline dev/com.ibm.ws.kernel.boot/resources/com/ibm/ws/kernel/boot/resources/LauncherMessages.nlsprops]:
```suggestion
error.invalidPhaseName=CWWKE0954E: The specified ({0}) checkpoint phase is empty or unknown.
```

**Review by dmuelle — CHANGES_REQUESTED** [review]
ID review, let me know if you have any questions, thanks

**Review by tbitonti — DISMISSED** [review]

File: dev/com.ibm.ws.kernel.boot/resources/com/ibm/ws/kernel/boot/resources/LauncherMessages.nlsprops (hunk 1 above)
Comment (jimblye) [reply dev/com.ibm.ws.kernel.boot/resources/com/ibm/ws/kernel/boot/resources/LauncherMessages.nlsprops]: The {0} is replaced by a value.   So in the displayed message, one would only see ( substituted text ).   I think it is helpful to separate the command line options and values which must be taken verbatim from the rest of the text which is translated.  For example: The ( 5mm ) value for the (--timeout) option is not valid.

**Comment by jimblye** [comment]
@dmuelle I committed your changes.   Ready for review again.

**Review by dmuelle — APPROVED** [review]
LGTM, thanks

**Review by tbitonti — APPROVED** [review]

**Comment by posmith** [comment]
L2 message review completed

**Comment by jimblye** [comment]
#libby

**Pull request merged**